tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs

```
@@ -203,17 +209,27 @@ public static string GetLatestStableVersion(string packageId)
             return version.ToString();
         }
 
-        /// <summary>
-        /// Returns the count of versions available for the given package
-        /// </summary>
-        public int GetVersionCount(string packageId, bool allowPreRelease = true)
+        public void VerifyVersionCount(string packageId, int expectedCount, bool allowPreRelease = true)
         {
             var repo = PackageRepositoryFactory.Default.CreateRepository(SourceUrl);
-            var packages = repo.FindPackagesById(packageId).ToList();
-            if (!allowPreRelease)
-                packages = packages.Where(item => item.IsReleaseVersion()).ToList();
-            return packages.Count;
+
+            Assert.True(VerifyWithRetry(
```

Will introducing retry logic here hide failures that we should be investigating?